### DISCUSSION TITLE
Is there a way to add profile image to the mention search list ?

### DISCUSSION THREAD

**Discussion by Dhrumilkp** [afconsult/quill-mention#372] [Q&A] [answered]
Is there a way to add profile image to mention search list...or any workarounds ?

**Comment by FrazeColder** [comment]
Push. Also interested in this. How can we modify the mention search list to add a profile image?

**Comment by MadSpindel** [comment]
Yes, you can use renderItem for that. What you need to do is to have something like "profileImageUrl" in the JSON search result, then in the renderItem create the HTML image with the url in item.profileImageUrl.

**Comment by FrazeColder** [comment]
@MadSpindel Can you please give us an exmaple on how do that?

I cannot find any example on the internet and just from your documentation I don't know how to do that.

**Comment by MadSpindel** [comment, marked as answer]
Sure, here is one quick example:

      var quill = new Quill("#editor", {
        placeholder: "Start by typing @ for mentions or # for hashtags...",
        modules: {
          mention: {
            allowedChars: /^[A-Za-z\sÅÄÖåäö]*$/,
            mentionDenotationChars: ["@", "#"],
            source: function(searchTerm, renderList, mentionChar) {
              let values;

              if (mentionChar === "@") {
                values = atValues;
              } else {
                values = hashValues;
              }

              if (searchTerm.length === 0) {
                renderList(values, searchTerm);
              } else {
                const matches = [];
                for (i = 0; i < values.length; i++)
                  if (
                    ~values[i].value
                      .toLowerCase()
                      .indexOf(searchTerm.toLowerCase())
                  )
                    matches.push(values[i]);
                renderList(matches, searchTerm);
              }
            },
            renderItem(item) {
              return `<img src="${item.profileImageUrl}">${item.value}`;
            },
          },
        },
      });

You see renderItem? I added img src with item.profileImageUrl. You need to make sure each item in the JSON Object list from the source function (which is sent as matches to renderList) has a profileImageUrl value with the correct url. You can use HTML in renderItem, so you can add divs and CSS styles etc. if you want.

**Comment by badspider7** [comment]
Hello,is there someone help,i try to render HTML,but it not work,render a string in <li></li>
![image](https://github.com/user-attachments/assets/ccff7fd4-0382-4d8f-82ed-ac62415355a9)
![image](https://github.com/user-attachments/assets/d5c58cbf-d989-478a-9868-a6d4ef0007ab)
